### PR TITLE
New UI

### PR DESCRIPTION
**Pull request by UnbearableBear**
La nouvelle UI est là !
Seul le composant bouton a été fait pour le moment.

Screenshot 2019-09-11 at 16 25 13
---------------------------------------
Screenshot 2019-09-11 at 16 25 22

J'ai aussi rajouté une règle à eslint concernant les packages afin qu'il rale lorsque des paquets non déclarés dans le packages.json sont importés et fonctionnent quand même grace au hoisting de yarn. Ca évitera des problèmes à l'avenir si on décide de split les packages pour de bon.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,11 +30,10 @@
...
     "@babel/preset-env": "7.5.5"
   },
-  "workspaces": [
-    "packages/*",
-    "packages/code-du-travail-data/dataset/!(fiches_service_public)",
-    "packages/code-du-travail-data/dataset/code_du_travail/eposeidon_script"
-  ],
+  "workspaces": {
+    "packages": ["packages/*", "packages/code-du-travail-data/dataset/*"],
+    "nohoist": ["**/react-ui", "**/react-ui/**"]
```
Comment (UnbearableBear) [inline package.json:35]: Impératif sinon docz crashe lors de son build car il ne va pas chercher react & co à la racine du proj
(pour le moment en tous cas)

File: packages/code-du-travail-data/dataset/fiches_service_public/package.json [hunk 2]
```diff
@@ -22,7 +22,6 @@
     "fetchData": "node fetchData.js",
     "start": "yarn backup && node index.js > fiches-sp-travail.json",
     "report": "node diff.js && node missingTheme.js",
-    "poststart": "yarn updateTheme && yarn report",
-    "test": "jest"
```
Comment (UnbearableBear) [inline packages/code-du-travail-data/dataset/fiches_service_public/package.json:26]: Il n'y a pas de tests dans ce repo

File: packages/code-du-travail-frontend/package.json [hunk 3]
```diff
@@ -16,20 +16,22 @@
...
     "debounce-promise": "^3.1.2",
+    "express": "^4.17.1",
     "express-sitemap-xml": "^1.1.0",
     "final-form": "^4.18.5",
     "final-form-arrays": "^3.0.1",
     "final-form-calculate": "^1.3.1",
     "isomorphic-unfetch": "^3.0.0",
     "memoizee": "^0.4.14",
     "next": "^9.0.5",
+    "prop-types": "^15.7.2",
```
Comment (lionelB) [inline packages/code-du-travail-frontend/package.json:34]: this is a dev-dep

File: packages/code-du-travail-frontend/package.json [hunk 4]
```diff
@@ -50,14 +52,15 @@
     "@babel/core": "^7.6.0",
     "@babel/preset-env": "^7.6.0",
     "@socialgouv/eslint-config-react": "^0.9.6",
+    "@testing-library/react": "^9.1.4",
```
Comment (lionelB) [inline packages/code-du-travail-frontend/package.json]: dev-dep too

File: packages/react-ui/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      format: "esm"
+    }
+  ],
+  external: ["prop-types", "react", "react-dom", "styled-components"],
+  plugins: [
+    babel(),
+    resolve({
+      mainFields: ["module", "jsnext", "jsnext:main", "main"]
+    }),
+    commonjs(),
```
Comment (lionelB) [inline packages/react-ui/rollup.config.js:24]: I thinks we can remove the commonjs() plugin here

File: packages/react-ui/src/icons/Cross.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
Comment (lionelB) [inline packages/react-ui/src/icons/Cross.js:1]: we don't use react-feather anymore ?

**Review by lionelB — COMMENTED** [review]
we would prefer a name tied to the project rather than react-ui :)
also, all the Theme related code to allow theme swapping should be add when we do a dark mode ;)

File: packages/react-ui/src/icons/Cross.js (hunk 6 above)
Comment (UnbearableBear) [reply packages/react-ui/src/icons/Cross.js:1]: I was supposed to start a new ui from scratch and they were no react-feather icons in the new design, I had all the SVG provided.
But we won't be doing the new design so I guess there is no need to spend too much time on this review.
Still you can use this branch to potentially play with rebass. You might be more successful implementing it in a fashion you like !

**Comment by UnbearableBear** [comment]
I think the only reason to use this new ui is the theming now that we dropped the new design. So I guess that removing theming code make this new ui concept deprecated for now. But I'm ok with keeping the old ui and close the two PR

File: packages/react-ui/src/icons/Cross.js (hunk 6 above)
Comment (lionelB) [reply packages/react-ui/src/icons/Cross.js:1]: Ok, is it possible to extract the eslint addition and fix you do ?

Comment (UnbearableBear) [reply packages/react-ui/src/icons/Cross.js:1]: Absolutely !

Comment (UnbearableBear) [reply packages/react-ui/src/icons/Cross.js:1]: I'll let the big PR be merged first the I'll take it.

**Pull request closed without merging**